### PR TITLE
Remove SQLAlchemy Roles features (supplanted by new Modeling & Data Filtering features)

### PR DESCRIPTION
**Pull request by gj**

### CODE REVIEW

**Comment by saolsen** [comment]
😭 💀

File: languages/python/oso/polar/host.py [hunk 1]
```diff
@@ -280,28 +280,15 @@ def to_polar(self, v):
...
-                # to allow Roles.role_allows to be called with unbound variables as
-                # arguments (only for sqlalchemy_oso)
-                # Because of this, we need to continue to send the repr for
-                # sqlalchemy_oso.roles.OsoRoles.Roles ONLY
-                if (
-                    "OsoRoles" in v.__qualname__
-                    and v.__module__ == "sqlalchemy_oso.roles"
-                ):
-                    repr_str = repr(v)
-                # END HACK
```
Comment (leina05) [inline languages/python/oso/polar/host.py:300]: 🙌🏻

File: languages/python/oso/polar/host.py [hunk 2]
```diff
@@ -280,28 +280,15 @@ def to_polar(self, v):
...
             val = {
                 "ExternalInstance": {
                     "instance_id": self.cache_instance(v, instance_id),
-                    "repr": repr_str,
+                    "repr": None,
```
Comment (leina05) [inline languages/python/oso/polar/host.py:291]: I think we can actually kill `repr` now--we were only keeping it around for roles. Can you confirm @gkaemmer?

File: languages/python/sqlalchemy-oso/requirements.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
 oso==0.20.0-beta
 SQLAlchemy>=1.3.17,<1.5
 packaging~=20.9
-dataclasses; python_version<"3.7"
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/requirements.txt:4]: 😱

File: languages/python/oso/polar/host.py (hunk 2 above)
Comment (leina05) [reply languages/python/oso/polar/host.py:291]: But not high priority, let's just add this as a task for next cycle.

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/compat.py [hunk 4]
```diff
@@ -15,6 +15,7 @@ def iterate_model_classes(base_or_registry):
     """Return an iterator of model classes that descend from a declarative base
     (SQLAlchemy 1.3 or 1.4) or exist in a registry (SQLAlchemy 1.4)."""
     try:  # 1.3 declarative base.
+        # TODO (dhatch): Not sure this is legit b/c it uses an internal interface?
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/compat.py:18]: @dhatch

**Review by leina05 — APPROVED** [review]
Alt Text

File: languages/python/oso/polar/host.py (hunk 2 above)
Comment (leina05) [reply languages/python/oso/polar/host.py:291]: I added this as a config follow-up, self-resolving.

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/compat.py (hunk 4 above)
Comment (gj) [reply languages/python/sqlalchemy-oso/sqlalchemy_oso/compat.py:18]: the comment still applies — the interface is still internal

File: languages/python/sqlalchemy-oso/requirements.txt (hunk 3 above)
Comment (gj) [reply languages/python/sqlalchemy-oso/requirements.txt:4]: not hurtin' anything, but no reason not to remove it now

**Pull request merged**